src/Tabs/Tabs.tsx

```diff
@@ -13,24 +13,103 @@ interface TabsProps {
     className?: string;
 }
 
-export const Tabs: React.FunctionComponent<TabsProps> = React.memo((props: TabsProps): React.ReactElement<void> => {
-    return (
-        <div className={"custom-tabs" + (props.className ? ` ${props.className}` : "")}>
-            <ul className="nav nav-tabs">
-                {props.list.map((item: TabsListItem, i: number) =>
-                    <li
-                        key={i}
-                        className={"nav-item" + (i === props.activeTab ? " active" : "") + (item.disabled ? " disabled" : "")}
-                    >
-                        <a
-                            className={"nav-link" + (i === props.activeTab ? " active" : "") + (item.disabled ? " disabled" : "")}
-                            onClick={() => { if (props.onClick && !item.disabled) { props.onClick(i); } }}
+interface TabsState {
+    localActiveTab: number;
+}
+
+export class Tabs extends React.Component<TabsProps, any> {
+    elementRefAnchor: Array<HTMLAnchorElement> = [];
+
+    constructor(props: TabsProps) {
+        super(props);
+        this.state = {
+            localActiveTab: props.activeTab
+        };
+
+        this.onKeyDown = this.onKeyDown.bind(this);
+        this.onClick = this.onClick.bind(this);
+    }
+
+    onClick(e: React.MouseEvent<HTMLAnchorElement>, item: TabsListItem, i: number): void {
+        if (this.props.onClick && !item.disabled) {
+            console.log("Its been called ", item);
+            this.props.onClick(i);
+        }
+    }
+
+    /**
+     * e: Event
+     */
+    onKeyDown(e: React.KeyboardEvent<HTMLAnchorElement>) {
```

Comment:
Since you're repeating `e.key.toLowerCase()` and `this.props.list` throughout the scope of this function I suggest for better readability to refactor like so:

```
const key = e.key.toLowerCase();
const { list } = this.props;
const { localActiveTab } = this.state;

if (localActiveTab < list.length && localActiveTab >= 0) {
...
}

 if ((key === "arrowleft" || key === "arrowdown") {
...
}


```